src/main/java/org/itsallcode/openfasttrace/importer/specobject/handler/SpecObjectsHandlerBuilder.java

```diff
@@ -47,19 +50,29 @@ public SpecObjectsHandlerBuilder(final String fileName, final String defaultDoct
 
     public CallbackContentHandler build()
     {
-        this.handler.addElementListener("specobject", elem -> {
-            this.listener.beginSpecificationItem();
-            this.listener.setLocation(this.fileName, elem.getLocation().getLine());
-            this.idBuilder = new SpecificationItemId.Builder() //
-                    .artifactType(this.defaultDoctype);
-            this.handler.pushDelegate(
-                    new SingleSpecObjectsHandlerBuilder(this.listener, this.idBuilder).build());
-        }, endElement -> {
-            this.listener.setId(this.idBuilder.build());
-            this.idBuilder = null;
-            this.listener.endSpecificationItem();
-        });
-
+        this.handler.addElementListener("specobject", this::handleStartElement,
+                this::handleEndElement);
         return this.handler;
     }
+
+    private void handleStartElement(final TreeElement elem)
+    {
+        this.listener.beginSpecificationItem();
+        this.locationBuilder = new Location.Builder() //
+                .path(this.fileName) //
+                .line(elem.getLocation().getLine());
+        this.idBuilder = new SpecificationItemId.Builder() //
+                .artifactType(this.defaultDoctype);
+        this.handler.pushDelegate(new SingleSpecObjectsHandlerBuilder(this.listener, this.idBuilder,
+                this.locationBuilder).build());
+    }
+
+    private void handleEndElement(final TreeElement elem)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this unused method parameter "elem". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)